build: always use correct path for mkhash

lemoer opened the pull request:
Before this commit, it was assumed that mkhash is in the PATH. While
this was fine for the normal build workflow, this led to some issues if

    make TOPDIR="$(pwd)" -C "$pkgdir" compile

was called manually. In most of the cases, I just saw warnings like this:

    make: Entering directory '/home/.../package/gluon-status-page'
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    [...]

While these were only warnings and the package still compiled sucessfully,
I also observed that some package even fail to build because of this.

After applying this commit, the location of mkhash is set to
$(STAGING_DIR_HOST)/bin/mkhash, which always works fine.

Signed-off-by: Leonardo Mörlein <me@irrelefant.net>

ynezz commented on include/depends.mk:
```diff
@@ -11,7 +11,7 @@
 
 DEP_FINDPARAMS := -x "*/.svn*" -x ".*" -x "*:*" -x "*\!*" -x "* *" -x "*\\\#*" -x "*/.*_check" -x "*/.*.swp" -x "*/.pkgdir*"
 
-find_md5=find $(wildcard $(1)) -type f $(patsubst -x,-and -not -path,$(DEP_FINDPARAMS) $(2)) -printf "%p%T@\n" | sort | mkhash md5
```
It would be probably better to create `MKHASH=$(STAGING_DIR_HOST)/bin/mkhash` variable probably in `rules.mk` and use that `$(MKHASH) md5` instead of this copy&pasta. While at it, do `git grep mkhash` and make the usage of that consistent.

ynezz commented on include/quilt.mk:
```diff
@@ -159,7 +159,7 @@ define Quilt/Template
 	}
 	@[ -n "$$$$(ls $(1)/patches/series)" -o \
-	   "$$$$(cat $(1)/patches/series | mkhash md5)" = "$$(sort $(1)/patches/series | mkhash md5)" ] || { \
+	   "$$$$(cat $(1)/patches/series | $(STAGING_DIR_HOST)/bin/mkhash md5)" = "$$(sort $(1)/patches/series | mkhash md5)" ] || { \
```
You see, you still have `mkhash` left here.

lemoer commented:
Thanks. I now used `git grep` and replaced all instances.

To do this in the perl and shell in `/scripts`, I had to export $MKHASH from the Makefile. Not sure if that's desired. Please let me know if that's undesired.

ynezz commented:
LGTM, squash that fixup commit.

lemoer committed:
build: introduce $(MKHASH)

Before this commit, it was assumed that mkhash is in the PATH. While
this was fine for the normal build workflow, this led to some issues if

    make TOPDIR="$(pwd)" -C "$pkgdir" compile

was called manually. In most of the cases, I just saw warnings like this:

    make: Entering directory '/home/.../package/gluon-status-page'
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    bash: line 1: mkhash: command not found
    [...]

While these were only warnings and the package still compiled sucessfully,
I also observed that some package even fail to build because of this.

After applying this commit, the variable $(MKHASH) is introduced. This
variable points to $(STAGING_DIR_HOST)/bin/mkhash, which is always the
correct path.

Signed-off-by: Leonardo Mörlein <me@irrelefant.net>

lemoer commented:
Thanks. I squashed it and adjusted the commit message.

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git
